File: data_structures/stack/js/stack.js
var Stack = (function(){
  function Stack() {
    this.dataStore = [];
    this.top = 0;
  }

  Stack.prototype.push = function(item) {
    this.dataStore[this.top] = item;
    this.top++;
  }

  Stack.prototype.pop = function() {
    this.top--;
    return this.dataStore[this.top];
  }

  Stack.prototype.display = function() {
    for(var i = 0; i < this.top; i++) {
      console.log(this.dataStore[i]);
    }
  }

  return Stack;
})();

module.exports = Stack;

File: data_structures/stack/rb/stack.rb
class Stack
  def initialize
    @data_store = []
    @top = 0
  end

  def push(item)
    @data_store[@top] = item
    @top += 1
  end

  def pop
    @top -= 1
    return @data_store[@top]
  end

  def display
    for i in 0...@top
      puts @data_store[i]
    end
  end
end
